interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -82,27 +82,50 @@ object TapirAdapter {
       oneOfVariantValueMatcher[CalibanBody.Single](customCodecJsonBody[ResponseValue].map(Left(_)) { case Left(value) =>
         value
       }) { case Left(_) => true },
+      oneOfVariantValueMatcher[CalibanBody.Single]({
+        stringBodyUtf8AnyFormat(codec.format(GraphqlResponseJson)).map(Left(_)) { case Left(value) => value }
+      }) { case Left(_) => true },
       oneOfVariantValueMatcher[CalibanBody.Stream[stream.BinaryStream]](
         streamTextBody(stream)(CodecFormat.Json(), Some(StandardCharsets.UTF_8)).toEndpointIO
           .map(Right(_)) { case Right(value) => value }
       ) { case Right(_) => true }
     )
 
   def buildHttpResponse[E, BS](
+    request: ServerRequest
+  )(
     response: GraphQLResponse[E]
   )(implicit
     streamConstructor: StreamConstructor[BS],
     responseCodec: JsonCodec[ResponseValue]
-  ): (MediaType, CalibanBody[BS]) =
+  ): (MediaType, StatusCode, CalibanBody[BS]) = {
+    def acceptsGqlJson = request.acceptsContentTypes.fold(
+      _ => false, // NOTE: 1st January 2025 this should be changed to `true`
```

Comment:
Maybe link to the reason for that? 😄

Reply:
Done. Also changed the description a bit because it wasn't very accurate